src/Microsoft.SqlTools.ServiceLayer/QueryExecution/ResultSet.cs

```diff
@@ -359,6 +362,9 @@ public Task<ExecutionPlan> GetExecutionPlan()
                     Columns = dataReader.Columns;
                     while (await dataReader.ReadAsync(cancellationToken))
                     {
+                        // Make sure we haven't cancelled
+                        cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Minor: this one probably isn't needed since it's just returned from a call that should be handling cancellation checks.